src/components/side_panel/pivot/editable_name/editable_name.ts

```diff
@@ -9,29 +9,67 @@ interface Props {
   onChanged: (name: string) => void;
 }
 
+interface State {
```

Comment:
The goal pf the other task I mentioned to you is to remove the `Edit` button, and to make the Pivot name into a `TextInput` so I think a lot of your changes in this file are unessecary since they will be deleted.

I think you can add the handling of Enter/Escape key directly inside the TextInputn and the rest of the commit is not needed. You can change the pivot renaming into a `TextInput` yourself without waiting for the other task if you want.